Lib/subprocess.py

```diff
@@ -412,6 +422,20 @@ def run(*popenargs, input=None, timeout=None, check=False, **kwargs):
             stdout, stderr = process.communicate()
             raise TimeoutExpired(process.args, timeout, output=stdout,
                                  stderr=stderr)
+        except KeyboardInterrupt:
+            if cleanup_timeout is None:
+                cleanup_timeout = 1.0
+                if timeout:
+                    remaining_timeout = timeout - (_time() - start_time)
+                    cleanup_timeout = max(0.0, remaining_timeout)
+
+            if cleanup_timeout == 0.0:
+                raise
+            try:
+                process.wait(timeout=cleanup_timeout)
+                raise
+            except TimeoutExpired:
+                raise KeyboardInterrupt
```

Comment:
Is it important to change the exception here? If so, why make a new KeyboardInterrupt exception rather than reraising the original one?

Reply:
>Is it important to change the exception here?

Yes, definitely.  The user hit `Ctrl+C`, so the expected behavior is to raise `KeyboardInterrupt`.

>If so, why make a new KeyboardInterrupt exception rather than reraising the original one?

Yeah, I asked myself this question, too.  I figured this was the clearest thing to write, and I also like the fact that, thanks to Python 3's automatic exception chaining, power-user could read the traceback carefully and see that the subprocess cleanup code timed out.

Anyway, I don't feel very strongly about it.  Open to suggestions.